src/deadline/job_attachments/download.py

```diff
@@ -741,6 +814,42 @@ def get_output_manifests_by_asset_root(
     return outputs
 
 
+def get_output_manifest_files_by_asset_root_with_last_modified(
+    s3_settings: JobAttachmentS3Settings,
+    output_manifest_paths: list[str],
+    session: Optional[boto3.Session] = None,
+) -> dict[str, list[Tuple[datetime, BaseAssetManifest]]]:
+    """
+    For a given output manifest path, gets a map from each root path to a corresponding list of
+    output manifests with their last modified timestamps.
+
+    Returns:
+        A dictionary mapping asset root paths to lists of tuples containing (last_modified, manifest)
+    """
+    outputs: DefaultDict[str, list[Tuple[datetime, BaseAssetManifest]]] = DefaultDict(list)
```

Comment:
To get the mirroring behavior I've suggested, you can initialize with:

```
outputs: list[tuple[str, datetime, BaseAssetManifest]] = [None] * len(output_manifest_paths)
```

Then make the parallel loop be `for index in range(len(output_manifest_paths))` so it can grab the correct item from `output_manifest_paths` and place the result into `outputs`.